server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -0,0 +1,321 @@
+                }
+            }
+        );
+    }
+
+    /**
+     * Registers new decommissioned attribute metadata in the cluster state
+     * <p>
+     * This method can be only called on the cluster-manager node. It tries to create a new decommissioned attribute on the master
+     * and if it was successful it adds new decommissioned attribute to cluster metadata.
+     * <p>
+     * This method should only be called once the eligible cluster manager node having decommissioned attribute is abdicated
+     *
+     * @param decommissionAttribute register decommission attribute in the metadata request
+     * @param listener              register decommission listener
+     */
+    private void registerDecommissionAttribute(
+        final DecommissionAttribute decommissionAttribute,
+        final ActionListener<ClusterStateUpdateResponse> listener
+    ) {
+        validateAwarenessAttribute(decommissionAttribute, getAwarenessAttributes());
+        clusterService.submitStateUpdateTask(
+            "put_decommission [" + decommissionAttribute + "]",
+            new ClusterStateUpdateTask(Priority.URGENT) {
+                @Override
+                public ClusterState execute(ClusterState currentState) throws Exception {
+                    logger.info("decommission request for attribute [{}] received", decommissionAttribute.toString());
+                    Metadata metadata = currentState.metadata();
+                    Metadata.Builder mdBuilder = Metadata.builder(metadata);
+                    DecommissionAttributeMetadata decommissionAttributeMetadata = metadata.custom(DecommissionAttributeMetadata.TYPE);
+                    ensureNoAwarenessAttributeDecommissioned(decommissionAttributeMetadata, decommissionAttribute);
+                    decommissionAttributeMetadata = new DecommissionAttributeMetadata(decommissionAttribute);
+                    mdBuilder.putCustom(DecommissionAttributeMetadata.TYPE, decommissionAttributeMetadata);
+                    return ClusterState.builder(currentState).metadata(mdBuilder).build();
+                }
+
+                @Override
+                public void onFailure(String source, Exception e) {
+                    // TODO - should we put the weights back to zone, since we weighed away the zone before we started registering the metadata
+                    // TODO - should we modify logic of logging for ease of debugging?
+                    if (e instanceof DecommissionFailedException) {
+                        logger.error(() -> new ParameterizedMessage("failed to decommission attribute [{}]", decommissionAttribute.toString()), e);
+                    } else if (e instanceof NotClusterManagerException) {
+                        logger.info(() -> new ParameterizedMessage(
+                            "cluster-manager updated while executing request for decommission attribute [{}]",
+                            decommissionAttribute.toString()), e);
```

Comment:
lets just log at one place. `cluster-manager updated` will be retried by `TransportDecommissionService` , so no need to log it separately here.

Reply:
Have put it as debug log for debugging purposes